modVM/src/org/aion/vm/ExecutionHelper.java

```diff
@@ -63,126 +69,78 @@ public Call(byte[] data, byte[] destination, byte[] value) {
         }
     }
 
-    private Set<Address> deleteAccounts = new HashSet<>();
-    private List<AionInternalTx> internalTxs = new ArrayList<>();
-    private List<Log> logs = new ArrayList<>();
-    private List<Call> calls = new ArrayList<>();
-
-    /**
-     * Create a new execution result.
-     */
-    public ExecutionHelper() {
-    }
-
-    /**
-     * Returns deleted accounts.
-     *
-     * @return
-     */
-    public List<Address> getDeleteAccounts() {
-        List<Address> result = new ArrayList<>();
-        for (Address acc : deleteAccounts) {
-            result.add(acc);
-        }
-        return result;
-    }
-
     /**
-     * Adds a deleted accounts.
+     * Adds address to the set of deleted accounts if address is non-null and is not already in the
+     * set.
      *
-     * @param address
+     * @param address The address to add to the deleted accounts.
      */
     public void addDeleteAccount(Address address) {
         deleteAccounts.add(address);
     }
 
     /**
-     * Adds a collections of deleted accounts.
+     * Adds the collection addresses to the set of deleted accounts if addresses is non-null.
      *
-     * @param addresses
+     * @param addresses The addressed to add to the set of deleted accounts.
      */
     public void addDeleteAccounts(Collection<Address> addresses) {
-        for (Address address : addresses) {
-            addDeleteAccount(address);
+        for (Address addr : addresses) {
+            if (addr != null) { deleteAccounts.add(addr); }
```

Comment:
nit/optional: I would have the if check in addDeleteAccount(Address) and then this loop just calls addDeleteAccount(Address)... not a big deal though

Reply:
I agree, I'll check with yulong about this tomorrow, the null checks may not even be necessary